### PR TITLE
Compound forks extended

### PR DESCRIPTION
**Pull request by 0xbe1**
🫡 Reviewers:

- The current change only works for moonwell, see [deployment](https://okgraph.xyz/?q=0xbe1%2Fmessari-moonwell-moonriver). To make the rest work, I want an LGTM to move on refactoring the rest.
- Main logic are in `addPosition` and `substractPosition` methods of src/mapping.ts

### CODE REVIEW

**Comment by this-username-is-taken** [comment]
can you put this in a separate folder? e.g. `compound-forks-ext`. This way we don't override our existing ones (we will maintain both at the same time).

**Comment by 0xbe1** [comment]
> can you put this in a separate folder? e.g. `compound-forks-ext`. This way we don't override our existing ones (we will maintain both at the same time). You can just copy paste the code.

Will do after we finish the pr review, otherwise it is impossible for reviewers to review the code.

**Comment by this-username-is-taken** [comment]
ah, good point

**Comment by this-username-is-taken** [comment]
where is this currently deployed?

**Comment by 0xbe1** [comment]
> where is this currently deployed?

updated description, check it out :)

File: subgraphs/compound-forks/protocols/moonwell/src/mapping.ts [hunk 1]
```diff
@@ -193,25 +212,97 @@ export function handleNewReserveFactor(event: NewReserveFactor): void {
 export function handleMint(event: Mint): void {
   let minter = event.params.minter;
   let mintAmount = event.params.mintAmount;
-  _handleMint(comptrollerAddr, minter, mintAmount, event);
+  let contract = CToken.bind(event.address);
+  let balanceOfUnderlyingResult = contract.try_balanceOfUnderlying(
+    event.params.minter
+  );
+  let balanceOfUnderlying = BigInt.fromI32(-1);
```
Comment (this-username-is-taken) [inline subgraphs/compound-forks/protocols/moonwell/src/mapping.ts:219]: Should make `BigInt.fromI32(-1)` a named constant

File: subgraphs/compound-forks/src/mapping.ts [hunk 2]
```diff
@@ -1640,6 +1870,217 @@ function getMarketDailySnapshotID(marketID: string, timestamp: i32): string {
...
+    position.balance = BIGINT_ZERO;
+    position.depositCount = 0;
+    position.withdrawCount = 0;
+    position.borrowCount = 0;
+    position.repayCount = 0;
+    position.liquidationCount = 0;
+    position.save();
+  }
+  position = position!;
+  if (balance.ge(BIGINT_ZERO)) {
```
Comment (this-username-is-taken) [inline subgraphs/compound-forks/src/mapping.ts:1925]: I think it'll be easier to understand if you test for equality with the named constant

File: subgraphs/compound-forks/src/mapping.ts [hunk 3]
```diff
@@ -1043,28 +1221,58 @@ function snapshotUsage(
...
       break;
     case EventType.Repay:
       dailySnapshot.dailyRepayCount += 1;
       break;
     case EventType.Liquidate:
       dailySnapshot.dailyLiquidateCount += 1;
+      if (newDAU) {
+        dailySnapshot.dailyActiveLiquidators += 1;
+      }
+      // TODO: handle liquidatee
```
Comment (this-username-is-taken) [inline subgraphs/compound-forks/src/mapping.ts:1262]: Does this need to be handled as well?

**Review by this-username-is-taken — COMMENTED** [review]
Looks great! Just the minor issues I pointed out. Then rename this to `compound-forks-ext` and we are good to go.

File: subgraphs/compound-forks/protocols/moonwell/src/mapping.ts (hunk 1 above)
Comment (0xbe1) [reply subgraphs/compound-forks/protocols/moonwell/src/mapping.ts:219]: cool let me name it `REVERTED` :)

**Pull request merged**